js/main.js

```diff
@@ -0,0 +1,191 @@
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var $searchResultsRaw = xhr.response.Search;
+    if (xhr.response.Error === 'Movie not found!') {
+      var $h1Search = document.querySelector('#h1-search');
+      $h1Search.textContent = 'Results';
+      var $pNoResult = document.querySelector('#p-no-results');
+      $pNoResult.className = 'white-text text-align-center';
+      $searchForm.elements.movieTitle.value = '';
+    } else {
+      data.view = 'results-view';
+      for (let i = 0; i < $searchResultsRaw.length; i++) {
+        var $result = renderSearchResult($searchResultsRaw[i]);
+        $resultsList.appendChild($result);
+      }
+      $searchView.className = 'hidden';
+      $resultsView.className = '';
+    }
+  });
+
+  xhr.send();
+}
+
+var $resultsList = document.querySelector('#results-list');
+
+function renderSearchResult(resultObj) {
+  var $liEl = document.createElement('li');
+  $liEl.className = 'result-card pad-1-bot marg-1-bot column-fourth';
+  $liEl.setAttribute('data-entry-id', resultObj.imdbID);
+  var $divEl1 = document.createElement('div');
+  $divEl1.className = 'search-img-container';
+  var $imgEl = document.createElement('img');
+  $imgEl.className = 'search-img';
+  if (resultObj.Poster === 'N/A') {
+    $imgEl.setAttribute('src', 'images/noposter.png');
+  } else {
+    $imgEl.setAttribute('src', resultObj.Poster);
+  }
+  var $iEl = document.createElement('i');
+  $iEl.className = 'fas fa-info-circle info-icon';
+  var $divEl2 = document.createElement('div');
+  $divEl2.className = 'pad-1-2-left-right text-align-center';
+  var $h3El = document.createElement('h3');
+  $h3El.className = 'result-title';
+  $h3El.textContent = resultObj.Title;
+
+  $divEl1.appendChild($imgEl);
+  $divEl1.appendChild($iEl);
+
+  $divEl2.appendChild($h3El);
+
+  $liEl.appendChild($divEl1);
+  $liEl.appendChild($divEl2);
+
+  return $liEl;
+}
+
+var $searchView = document.querySelector('div#search-view');
+var $resultsView = document.querySelector('div#results-view');
+
+var $navBar = document.querySelector('#nav-bar');
+
+$navBar.addEventListener('click', swapViewNav);
+
+function swapViewNav(event) {
+  if (event.target.matches('#nav-search')) {
+    data.view = 'search-view';
+    $searchView.className = '';
+    $resultsView.className = 'hidden';
+    $searchForm.elements.movieTitle.value = '';
+    var $h1Search = document.querySelector('#h1-search');
+    $h1Search.textContent = 'Search';
+    var $pNoResult = document.querySelector('#p-no-results');
+    $pNoResult.className = 'hidden white-text text-align-center';
+    $searchForm.elements.movieTitle.value = '';
+    var $liElementList = document.querySelectorAll('li');
+    for (let i = 0; i < $liElementList.length; i++) {
+      $liElementList[i].remove();
+    }
+  }
+}
+
+var $modalMovieDetails = document.querySelector('#modal-movie-details');
+$resultsView.addEventListener('click', showMovieDetails);
+var $closestLi = null;
```

Comment:
Why is this global? It should be inside whichever function it is being used for.

Reply:
Fixed it!